app/helpers/user_helper.rb

```diff
@@ -10,11 +10,18 @@ def current_user?(user)
   def user_info(user)
     return if current_user?(user)
 
+    # rubocop:disable Lint/UselessAssignment
     request = Friendship.find_by(user_id: user.id, friend_id: current_user.id, confirmed: false)
-    if request.nil?
+    # rubocop:enable Lint/UselessAssignment
+    if current_user.friends.include?(user)
       render partial: 'user', locals: { user: user }
+    elsif current_user.pending_friends.include?(user)
+      render partial: 'users/pending', locals: { user: user }
+    elsif current_user.friend_requests.include?(user)
+      request = Friendship.find_by(user_id: user.id, friend_id: current_user.id, confirmed: false)
+      render partial: 'friendships/confirm_decline', locals: { request: request }
     else
-      confirm_decline(request)
+      friend_request(user)
```

Comment:
- [ ] I can still see a button to request friendship even after requesting, this problem allows a user to send requests to the same user more than once, this can be fixed.
- [ ] A current user should not have access to add friend button. See the images below.

![Current-user](https://user-images.githubusercontent.com/53124548/133873472-86bc6b91-b772-4482-aba8-5a86dff59fd6.png)
![after](https://user-images.githubusercontent.com/53124548/133873480-74a4bb89-28ec-45c5-8b3c-2c980bc94c93.png)